src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -162,7 +162,7 @@ static HYD_status list_to_nodes(char *str)
         gpattern[0] += gmatch[0][0].rm_eo;
     }
 
-    /* match new group-0 pattern: (,|^)(h|h-)([00-12,14] | 00-12 | 14)(,|$) */
+    /* match new group-0 pattern: (,|^)(h|h-|h00|h00-)([00-12,14] | [00-12] | 14)(,|$) */
     while (*gpattern[0] && regexec(&gmatch_new[0], gpattern[0], MAX_GMATCH, gmatch[0], 0) == 0) {
```

Comment:
I find it is more helpful to copy the exact regex pattern here as the comment.  The half-comment/half-regex is quite confusing.

(If you decide to change, the comment applies to other similar places as well.)

Reply:
Actually these comments are useful to see how the parsing is performed by the code and it helps me to understand what I have done when I go back to the code after a long time. Replacing these with the REGEX would not provide any additional information.